delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Permissions.scala

```diff
@@ -3,18 +3,20 @@ package ch.epfl.bluebrain.nexus.delta.sdk
 import java.time.Instant
 
 import cats.effect.Clock
+import ch.epfl.bluebrain.nexus.delta.sdk.model.Envelope
 import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
 import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.PermissionsCommand._
 import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.PermissionsEvent._
 import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.PermissionsRejection._
 import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.PermissionsState.{Current, Initial}
 import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions._
-import monix.bio.{IO, UIO}
+import fs2.Stream
+import monix.bio.{IO, Task, UIO}
 
 /**
   * Operations pertaining to managing permissions.
   */
-trait Permissions {
+trait Permissions[Offset] {
```

Comment:
It has to be a generic type? Cannot be an abstract type? We are creating a lot of unnecessary noise having it as a type parameter (like `Permissions[_]` or `Permissions[Nothing]`)